python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -0,0 +1,153 @@
+                  for i in range(1, 4)]
+        for n in range(0, len(vboost)):
+            print("vboost" + str(n) + " is: " + str(vboost[n]) + " mV")
+
+        vbatt = int(read_state(self, "gomspace.vbatt"))
+        if vbatt < 6000 or vbatt > 8400:
+            print(
+                "Vbatt is out of expected range [6000, 8400] mV at: " + str(vbatt))
+
+        curin = [int(read_state(self, "gomspace.curin.output" + str(i)))
+                 for i in range(1, 4)]
+        for n in range(0, len(curin)):
+            print("curin" + str(n) + " is: " + str(curin[n]) + " mA")
+        cursun = int(read_state(self, "gomspace.cursun"))
+        print("cursun is: " + str(cursun) + " mA")
+        cursys = int(read_state(self, "gomspace.cursys"))
+        print("cursys is: " + str(cursys) + " mA")
+
+        curout = [int(read_state(self, "gomspace.curout.output" + str(i)))
+                  for i in range(1, 7)]
+        for n in range(0, len(curout)):
+            print("curout" + str(n) + " is: " + str(curout[n]) + " mA")
+
+        output = [bool(read_state(self, "gomspace.output.output" + str(i)))
+                  for i in range(1, 7)]
+
+        for n in range(0, len(output)):
+            out_n = output[n]
+            # checked umbilical for which outputs should be 0 and 1:
+            # 1-5 are all 5V, 6 is 3.3V
+            if out_n is False:
+                print("Output-" + n + " is not on")
+
+        wdt_i2c_time_left = int(read_state(self, "gomspace.wdt_i2c_time_left"))
+        if wdt_i2c_time_left < 99:
+            print("wdt_i2c_time_left is less than 99 seconds at: " +
+                  str(wdt_i2c_time_left))
+
+        counter_wdt_i2c = read_state(self, "gomspace.counter_wdt_i2c")
+        print("counter_wdt_i2c is: " + str(counter_wdt_i2c))
+        counter_boot = read_state(self, "gomspace.counter_boot")
+        print("counter_wdt_i2c is: " + str(counter_boot))
+
+        temp = [int(read_state(self, "gomspace.temp.output" + str(i)))
+                for i in range(1, 5)]
+        for n in range(0, len(temp)):
+            temp_n = temp[n]
+            if temp_n < 20 or out_n > 25:
+                print("Temp-" + str(n) +
+                      " is out of room temperature range [20, 25] degC at: " + str(temp_n))
+
+        bootcause = read_state(self, "gomspace.bootcause")
+        print("bootcause is: " + str(bootcause))
+        battmode = read_state(self, "gomspace.battmode")
+        print("battmode is: " + str(battmode))
+        pptmode = read_state(self, "gomspace.pptmode")
+        print("pptmode is: " + str(pptmode))
+
+        # writable fields
+        power_cycle_output_cmd = [self.str_to_bool(read_state(self, "gomspace.power_cycle_output" + str(i) + "_cmd"))
+                                  for i in range(1, 7)]
+        cycle_no_init = int(read_state(self, "pan.cycle_no"))
+        cycle_no = cycle_no_init
+        print(power_cycle_output_cmd)
+        # wait for power cycling to be off
```

Comment:
The power cycle command statefield is a command that will turn off and then turn on the gomspace outputs. When the gomspace control task is initialized, all the power cycle output commands are all set to false. So the first while loop here is kind of unnecessary because the power cycle commands should be initialized to false anyway. 

You can start a power cycle for specific output by setting the power cycle command statefield for that output to true. If a power cycle command is true, that means that the gomspace is currently power cycling that output, but it _doesn't_ tell you whether or not that output is currently on or off. You can check whether or not the output is currently on or off by reading the gomspace.output.outputX statefields. 

So you would have to set all the power cycle output commands to true to start a power cycle. And _then_ you would wait for the outputs to turn off within 30-60 seconds by reading the gomspace.output.outputX statefields. 

Once they are all off, then the gomspace control task should then start to turn the outputs currently power cycling on again. Then this is where you again would wait around 30-60 seconds again for all the outputs turn on.